app/components/channel_link/channel_link.test.js

```diff
@@ -58,18 +65,46 @@ describe('ChannelLink', () => {
     });
 
     test('should call props.actions and onChannelLinkPress on handlePress', () => {
+        const actions = {...baseProps.actions, joinChannel: jest.fn()};
         const wrapper = shallow(
-            <ChannelLink {...baseProps}/>
+            <ChannelLink
+                {...baseProps}
+                actions={actions}
+            />,
+            {context: {intl: {formatMessage}}},
         );
 
         const channel = channelsByName.firstChannel;
-        wrapper.setState({channel});
         wrapper.instance().handlePress();
-        expect(baseProps.actions.handleSelectChannel).toHaveBeenCalledTimes(1);
-        expect(baseProps.actions.handleSelectChannel).toBeCalledWith(channel.id);
-        expect(baseProps.actions.setChannelDisplayName).toHaveBeenCalledTimes(1);
-        expect(baseProps.actions.setChannelDisplayName).toBeCalledWith(channel.display_name);
+        expect(actions.handleSelectChannel).toHaveBeenCalledTimes(1);
+        expect(actions.handleSelectChannel).toBeCalledWith(channel.id);
+        expect(actions.setChannelDisplayName).toHaveBeenCalledTimes(1);
+        expect(actions.setChannelDisplayName).toBeCalledWith(channel.display_name);
         expect(baseProps.onChannelLinkPress).toHaveBeenCalledTimes(1);
         expect(baseProps.onChannelLinkPress).toBeCalledWith(channel);
+
+        expect(actions.joinChannel).not.toBeCalled();
+    });
+
+    test('should call props.actions.joinChannel on handlePress when user is not member of such channel', async () => {
+        const newChannelName = 'thirdChannel';
+        const thirdChannel = {id: 'channel_id_3', name: 'thirdChannel', display_name: 'thirdChannel', team_id: 'current_team_id'};
+        const joinChannel = jest.fn().mockResolvedValue({data: {channel: thirdChannel}});
```

Comment:
Similarly, you can just call `baseProps.joinChannel.mockResolvedValue` instead of defining a new `joinChannel` here

Reply:
Not sure if you mean is something like:
```javascript
        const newChannelName = 'thirdChannel';
        const thirdChannel = {id: 'channel_id_3', name: 'thirdChannel', display_name: 'thirdChannel', team_id: 'current_team_id'};
        const channelMentions = {thirdChannel: {display_name: 'thirdChannel'}};
        const newChannelsByName = Object.assign({}, channelMentions, channelsByName);
        baseProps.actions.joinChannel.mockResolvedValue({data: {channel: thirdChannel}});
        baseProps.channelsByName = newChannelsByName;
        baseProps.channelName = newChannelName;

        const wrapper = shallow(
            <ChannelLink {...baseProps}/>,
            {context: {intl: {formatMessage}}},
        );

        await wrapper.instance().handlePress();
        expect(baseProps.actions.joinChannel).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.joinChannel).toBeCalledWith('current_user_id', 'current_team_id', null, newChannelName);
```
If yes, I prefer defining a new one for `newProps`.

Reply:
Yeah, that makes sense now that I think about it. It's nicer to use the shorter name for the action as well anyway.